### PR TITLE
Config class for SimpleSpanProcessor

### PR DESCRIPTION
**Pull request by thisthat**

### CODE REVIEW

File: sdk/src/main/java/io/opentelemetry/sdk/trace/export/SimpleSpansProcessor.java [hunk 1]
```diff
@@ -79,50 +106,137 @@ public void forceFlush() {
...
      */
-    public Builder reportOnlySampled(boolean sampled) {
-      this.sampled = sampled;
-      return this;
+    public static Config getDefault() {
+      return newBuilder().build();
     }
 
-    // TODO: Add metrics for total exported spans.
-    // TODO: Consider to add support for constant Attributes and/or Resource.
```
Comment (bogdandrutu) [inline sdk/src/main/java/io/opentelemetry/sdk/trace/export/SimpleSpansProcessor.java:115]: Can you please leave these todos? Or create issue if you want.

File: sdk/src/main/java/io/opentelemetry/sdk/common/export/ConfigBuilder.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.Maps;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+public abstract class ConfigBuilder<T> {
```
Comment (jkwatson) [inline sdk/src/main/java/io/opentelemetry/sdk/common/export/ConfigBuilder.java:28]: Please add javadoc about what `<T>` is here.

File: sdk/src/main/java/io/opentelemetry/sdk/trace/export/SimpleSpansProcessor.java (hunk 1 above)
Comment (thisthat) [reply sdk/src/main/java/io/opentelemetry/sdk/trace/export/SimpleSpansProcessor.java:115]: Could you please elaborate more on the second TODO? So I can provide a comprehensive description for the issue (and take it :wink:)

File: sdk/src/main/java/io/opentelemetry/sdk/common/export/ConfigBuilder.java [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+      }
+    },
+    ENV_VAR {
+      @Override
+      public String normalize(@Nonnull String key) {
+        return key.toLowerCase().replace("_", ".");
+      }
+    };
+
+    public abstract String normalize(@Nonnull String key);
```
Comment (carlosalberto) [inline sdk/src/main/java/io/opentelemetry/sdk/common/export/ConfigBuilder.java:50]: What's the rationale behind having `normalize` on both the naming convention enum and the `ConfigBuilder` itself?

File: sdk/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpansProcessor.java [hunk 4]
```diff
@@ -151,126 +147,6 @@ public void forceFlush() {
...
-   */
-  public static Builder newBuilder(SpanExporter spanExporter) {
-    return new Builder(spanExporter);
-  }
-
-  /** Builder class for {@link BatchSpansProcessor}. */
-  public static final class Builder {
-
-    private static final long SCHEDULE_DELAY_MILLIS = 5000;
-    private static final int MAX_QUEUE_SIZE = 2048;
```
Comment (carlosalberto) [inline sdk/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpansProcessor.java:169]: What happened to these default values? Doing a quick pass, I don't see them anywhere and don't know whether they exist anymore in this PR.

Comment (thisthat) [reply sdk/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpansProcessor.java:169]: They have been moved to the [`Config` class](https://github.com/dynatrace-oss-contrib/opentelemetry-java/blob/config/sdk/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpansProcessor.java#L315)

File: sdk/src/main/java/io/opentelemetry/sdk/common/export/ConfigBuilder.java (hunk 3 above)
Comment (thisthat) [reply sdk/src/main/java/io/opentelemetry/sdk/common/export/ConfigBuilder.java:50]: This method is abstract so each `NamingConvention` has to implement its normalization. It is only available inside the enum.

Comment (carlosalberto) [reply sdk/src/main/java/io/opentelemetry/sdk/common/export/ConfigBuilder.java:50]: Oooh, my bad. I had read it as being *outside* the enum.

**Comment by carlosalberto** [comment]
Good effort, thanks for refactoring this code! Looking hot!

**Pull request merged**